web/src/pages/apd/new/ApdNew.js

```diff
@@ -0,0 +1,210 @@
+  const setYears = value => {
+    try {
+      let index = years.indexOf(value);
+      if (index < 0) {
+        let newYears = years.push(value);
+        initialState.years = newYears;
+      } else {
+        years.splice(index);
+      }
+    } catch {
+      err => console.log(err);
+    }
+  };
+
+  const validate = createApdSchema.validate(initialState);
+
+  return (
+    <Fragment>
+      <div className="ds-u-margin--7 ds-u-padding--7 ds-u-padding-right--12">
+        <h1 className="ds-h2">Create a New Advanced Planning Document (APD)</h1>
+        <div className="ds-u-padding-bottom--1 ds-u-border-bottom--2">
+          Complete all the fields below to create your APD.
+        </div>
+        <h2 className="ds-h3">What type of APD are you creating?</h2>
+        <Alert
+          variation="warn"
+          className="ds-u-margin-y--3 ds-u-margin-right--7"
+        >
+          <p className="ds-c-alert__text">
+            This selection cannot be changed after creating a new APD.
+          </p>
+        </Alert>
+        <Controller
+          name="apdType"
+          control={control}
+          render={({ field: { onChange, value, ...props } }) => (
+            <ChoiceList
+              {...props}
+              type="radio"
+              choices={[
+                {
+                  label: 'HITECH IAPD',
+                  value: 'hitech',
+                  checked: value === 'hitech'
+                },
+                {
+                  label: 'MMIS IAPD',
+                  value: 'mmis',
+                  checked: value === 'mmis'
+                }
+              ]}
+              onChange={e => {
+                onChange(e);
+                setApdType(e.target.value);
+              }}
+            />
+          )}
+        />
+        {(apdType === 'mmis' || apdType === 'hitech') && (
+          <div>
+            <Controller
+              name="name"
+              control={control}
+              value=""
+              render={({ field: { ...props } }) => (
+                <TextField
+                  {...props}
+                  label="APD Name"
+                  className="remove-clearfix"
+                  onBlur={e => {
+                    setName(e.target.value);
+                    trigger();
+                  }}
+                />
+              )}
+            />
+            <Controller
+              name="years"
+              control={control}
+              label="Federal Fiscal Year (FFY)"
+              render={({ field: { onChange, ...props } }) => (
+                <ChoiceList
+                  {...props}
+                  choices={yearChoices}
+                  labelClassName="ds-u-margin-bottom--1"
+                  hint="Choose the federal fiscal year(s) this APD covers."
+                  type="checkbox"
+                  value=""
+                  onChange={e => {
+                    onChange(e);
+                    setYears(e.target.value);
+                  }}
```

Comment:
for some of the elements, especially the ones from the design system you can't just do the {...props} thing, not sure why because it seems like it should work, you need to pull out onBlur and do onBlur={onBlur) in the ChoiceList props, also try setting onComponentBlur={onBlur}

Reply:
if you are using the setYears above and it's now calling the set state of react-hook-forms, then you don't need to call onChange any more, but you should try not using setYears at all and see if react-hook-forms will keep track of what's checked in an array or how it maintains it, print off the value of years somewhere using getValues("years")